### PR TITLE
[extractor/vevo] Fix extractor

### LINKED ISSUE #3809: tv.vevo.com
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Brazil

### Example URLs

https://tv.vevo.com/watch/artist/janet-jackson/US0450100550
https://tv.vevo.com/watch/artist/rick-astley/GB1108800260

### Description

The default Vevo extractor can't download some videos that are available in tv.vevo.com

### Verbose log

```shell
[debug] Command-line config: ['-vU', 'https://tv.vevo.com/watch/artist/janet-jackson/US0450100550']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Python version 3.10.4 (CPython 64bit) - Windows-10-10.0.22000-SP0
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg git-2022-03-27-c2d1597-ffmpeg-windows-build-helpers (fdk,setts), ffprobe git-2022-03-27-c2d1597-ffmpeg-windows-build-helpers
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[debug] [generic] Extracting URL: https://tv.vevo.com/watch/artist/janet-jackson/US0450100550
[generic] US0450100550: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] US0450100550: Downloading webpage
[generic] US0450100550: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://tv.vevo.com/watch/artist/janet-jackson/US0450100550
Traceback (most recent call last):
  File "C:\Users\glomatico\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1395, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\glomatico\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1465, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\glomatico\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 642, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\glomatico\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\extractor\generic.py", line 4031, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://tv.vevo.com/watch/artist/janet-jackson/US0450100550
```

### LINKED ISSUE #3919: An extractor error has occurred. (caused by KeyError('quality'));
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

The vevo URLs are not playable in the browser, so I was not able to check this. But this error appears, and I don't know how to handle it. Can you help out?

### Verbose log

```shell
./yt-dlp --verbose -F http://www.vevo.com/watch/ES1022200025
[debug] Command-line config: ['--verbose', '-F', 'http://www.vevo.com/watch/ES1022200025']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (zip)
[debug] Python version 3.9.12 (CPython 64bit) - macOS-12.4-x86_64-i386-64bit
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: avprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: avconv -bsfs
[debug] exe versions: rtmpdump 2.4
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] [Vevo] Extracting URL: http://www.vevo.com/watch/ES1022200025
[Vevo] Retrieving oauth token
[Vevo] ES1022200025: Downloading api video info
[Vevo] ES1022200025: Downloading video versions info
ERROR: ES1022200025: An extractor error has occurred. (caused by KeyError('quality')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/Users/michaelhornke/Development/youtube-song-downloader/youtube-dl/bin/./yt-dlp/yt_dlp/extractor/common.py", line 642, in extract
    ie_result = self._real_extract(url)
  File "/Users/michaelhornke/Development/youtube-song-downloader/youtube-dl/bin/./yt-dlp/yt_dlp/extractor/vevo.py", line 152, in _real_extract
    'format_id': 'http-%s-%s' % (version, video_version['quality']),
KeyError: 'quality'
```

### PR DESCRIPTION
**Pull request by Lesmiscore**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Closes #3809
Closes #3919

<details>
<summary>pre-table output part</summary>

```
$ ./yt-dlp.sh -vF vevo:US0450100550
[debug] Command-line config: ['-vF', 'vevo:US0450100550']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 1cd6cba30
[debug] Python version 3.10.4 (CPython 64bit) - Linux-5.15.0-30-generic-x86_64-with-glibc2.35
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, pyxattr-0.7.2, secretstorage-3.3.1, sqlite3-2.6.0, websockets-9.1
[debug] Proxy map: {}
[debug] [Vevo] Extracting URL: vevo:US0450100550
[Vevo] Retrieving oauth token
[Vevo] US0450100550: Downloading api video info
[Vevo] US0450100550: Downloading video versions info
[Vevo] US0450100550: Downloading level3 m3u8 information
[Vevo] US0450100550: Downloading level3 MPD information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
```

</details>

```
[info] Available formats for US0450100550:
ID               EXT RESOLUTION FPS │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC      ABR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
http-level3-low  mp4 256x144        │ ~11.09MiB  328k http   │ h264         200k aac        128k
dash-level3-v1   mp4 416x234     25 │ ~ 1.90MiB   56k http   │ avc1.4D401F   56k video only      DASH video, mp4_dash
dash-level3-v2   mp4 416x234     25 │ ~ 5.80MiB  171k http   │ avc1.4D401F  171k video only      DASH video, mp4_dash
hls-level3-219   mp4 416x234     25 │ ~ 7.42MiB  219k m3u8_n │ avc1.4d001f  219k mp4a.40.2    0k
hls-level3-363   mp4 416x234     25 │ ~12.28MiB  363k m3u8_n │ avc1.4d001f  363k mp4a.40.2    0k
dash-level3-v3   mp4 480x270     25 │ ~11.54MiB  341k http   │ avc1.4D401F  341k video only      DASH video, mp4_dash
hls-level3-551   mp4 480x270     25 │ ~18.66MiB  551k m3u8_n │ avc1.4d001f  551k mp4a.40.2    0k
http-level3-None mp4 640x360        │ ~24.62MiB  728k http   │ h264         600k aac        128k
dash-level3-v4   mp4 640x360     25 │ ~14.40MiB  425k http   │ avc1.4D401F  425k video only      DASH video, mp4_dash
dash-level3-v5   mp4 640x360     25 │ ~22.95MiB  678k http   │ avc1.4D401F  678k video only      DASH video, mp4_dash
hls-level3-742   mp4 640x360     25 │ ~25.12MiB  742k m3u8_n │ avc1.4d001f  742k mp4a.40.2    0k
hls-level3-944   mp4 768x432     25 │ ~31.95MiB  944k m3u8_n │ avc1.4d001f  944k mp4a.40.2    0k
http-level3-high mp4 1280x720       │ ~71.96MiB 2128k http   │ h264        2000k aac        128k
```

Maybe:
- [x] Support URL formats like `https://tv.vevo.com/watch/artist/janet-jackson/US0450100550`
- [x] Copy tests from youtube-dl

### CODE REVIEW

File: yt_dlp/extractor/vevo.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def _real_extract(self, url):
 
                 formats.append({
                     'url': version_url,
-                    'format_id': 'http-%s-%s' % (version, video_version['quality']),
+                    'format_id': f'http-{version}-{video_version.get("quality")}',
```
Comment (dirkf) [inline yt_dlp/extractor/vevo.py:152]: You can get the missing quality from the URL, which is already being analysed quite fully:
```diff
---  old/yt-dlp/yt_dlp/extractor/vevo.py
+++ new/yt-dlp/yt_dlp/extractor/vevo.py
@@ -138,6 +138,7 @@
                     fatal=False))
             else:
                 m = re.search(r'''(?xi)
+                    _(?P<quality>[a-z0-9]+)
                     _(?P<width>[0-9]+)x(?P<height>[0-9]+)
                     _(?P<vcodec>[a-z0-9]+)
                     _(?P<vbr>[0-9]+)
@@ -149,7 +150,7 @@
 
                 formats.append({
                     'url': version_url,
-                    'format_id': 'http-%s-%s' % (version, video_version['quality']),
+                    'format_id': 'http-%s-%s' % (version, video_version.get('quality', m.group('quality'))),
                     'vcodec': m.group('vcodec'),
                     'acodec': m.group('acodec'),
                     'vbr': int(m.group('vbr')),
```

Also the yt-dl extractor has tests, missing in this module: consider pulling them in if they work.

Comment (Lesmiscore) [reply yt_dlp/extractor/vevo.py:152]: Blame on youtube-dl repo suggests me that tests were deleted in either youtube-dlc or yt-dlp... why

**Comment by Lesmiscore** [comment]
@pukkandan Do you think tests should be brought back to yt-dlp?
They aren't currently in yt-dlp's vevo.py

https://github.com/ytdl-org/youtube-dl/blame/master/youtube_dl/extractor/vevo.py#L41-L151

File: yt_dlp/extractor/vevo.py [hunk 2]
```diff
@@ -149,7 +151,7 @@ def _real_extract(self, url):
 
                 formats.append({
                     'url': version_url,
-                    'format_id': 'http-%s-%s' % (version, video_version['quality']),
+                    'format_id': f'http-{version}-{video_version.get("quality") or m.group("quality"))}',
```
Comment (pukkandan) [inline yt_dlp/extractor/vevo.py:154]:
```suggestion
                    'format_id': f'http-{version}-{video_version.get("quality") or m.group("quality")}',
```

**Pull request merged**